### ISSUE TITLE
Found 0 sections and 0 lectures on this page on some courses

### ISSUE THREAD

**Issue by Mahesha999** [coursera-dl/coursera-dl#512]
### Subject: Getting error message like below for some classes 
```
>coursera-dl -u <email> -p <password> -sl en -pl --clear-cache --debug --about --path C:\Users\user\Downloads\Video\coursera introstats-001
root[main] coursera_dl version 0.6.0
root[main] Downloading class: introstats-001
root[download_class] Downloading old style class introstats-001
root[get_cookies_from_cache] Trying to get cookies from the cache.
root[load_cookies_file] Loading cookie file C:\Users\user\AppData\Local\Temp\crrma_coursera_dl_cache\cookies\<emailaddr>.txt into memory.
root[get_cookies_from_cache] Could not load cookies from the cache.
root[login] Initiating login.
root[login] There were no .coursera.org cookies to be cleared.
requests.packages.urllib3.connectionpool[_new_conn] Starting new HTTPS connection (1): class.coursera.org
requests.packages.urllib3.connectionpool[_make_request] "GET /introstats-001 HTTP/1.1" 302 None
root[login] Forging cookie header: csrftoken=HQI7seaTL6U5aQDtueyN; csrf2_token_qi70UeRV=WqkjphGau7MzsnonZhsBOgu4.
requests.packages.urllib3.connectionpool[_new_conn] Starting new HTTPS connection (1): www.coursera.org
requests.packages.urllib3.connectionpool[_make_request] "POST /api/login/v3 HTTP/1.1" 200 None
root[login] Logged in on coursera.org.
requests.packages.urllib3.connectionpool[_new_conn] Starting new HTTPS connection (1): class.coursera.org
requests.packages.urllib3.connectionpool[_make_request] "GET /introstats-001/auth/auth_redirector?type=login&subtype=normal HTTP/1.1" 302 None
requests.packages.urllib3.connectionpool[_make_request] "GET /course/introstats HTTP/1.1" 200 None
root[down_the_wabbit_hole] Following https://class.coursera.org/introstats-001/auth/auth_redirector?type=login&subtype=normal to authenticate on class.coursera.org.
root[down_the_wabbit_hole] Exiting "deep" authentication.
root[get_authentication_cookies] Found authentication cookies.
root[download_about] About file to be written to: C:\Users\user\Downloads\Video\coursera\introstats-001-about.json
root[download_old_style_class] Subtitle unavailable in specified language
root[get_syllabus_url] Using index mode with page: https://class.coursera.org/introstats-001/lecture/index
requests.packages.urllib3.connectionpool[_make_request] "GET /introstats-001/lecture/index HTTP/1.1" 301 None
requests.packages.urllib3.connectionpool[_make_request] "GET /introstats-001/lecture HTTP/1.1" 302 None
requests.packages.urllib3.connectionpool[_make_request] "GET /course/introstats HTTP/1.1" 200 None
root[get_old_style_syllabus] Downloaded https://class.coursera.org/introstats-001/lecture/index (1175500 bytes)
root[parse_old_style_syllabus] Found 0 sections and 0 lectures on this page
root[parse_old_style_syllabus] The cookies file may be invalid, please re-run with the `--clear-cache` option.
```
I tried this for following classes:
- stats1-2012-001					(Statistics One)
- stats1-002
- stats1-003
- introstats-001					(Statistics: Making Sense of Data)			
- introstats-002
- introstats-003
- pdstatistics-001				(Passion Driven Statistics)

Am able to successfully download pgm (Probabilistic Graphical Model) class.

### Environment
* Windows 10 and Ubuntu 14.04 LTS
* Python version: 3.4
* coursera-dl version: 0.6.0

Am new to everything here (python, pip, coursera-dl). I was able to download pgm (Probabilistic Graphical Models) class successfully few hours ago. After that I started getting above messages for all courses I tried. I tried downloading on both Windows and Ubuntu. I did not installed dependencies manually. I installed it with
```
apt-get install python3 python3-pip
pip3 install coursera-dl 
```
So I have following (mostly latest) versions of dependencies:
 - beautifulsoup4 (4.4.1) on both Windows and Ubuntu
 - html5lib (1.0b2) on Windows and html5lib (1.0b8) on Ubuntu
 - keyring (9.0) on both Windows and Ubuntu
 - requests (2.9.1) on Windows and (2.10.0) on Ubuntu
 - six (1.10.0) on Windows and (1.5.2) on Ubuntu
 - urllib3 (1.16) on both Windows and Ubuntu
 - pyasn1 (0.1.9) on both Windows and Ubuntu

So I guess comparing against requirement.txt these look ok.

**Comment by balta2ar** [comment]
@Mahesha999 Please see this section: https://github.com/coursera-dl/coursera-dl/blob/master/README.md#found-0-sections-and-0-lectures-on-this-page

**Comment by Mahesha999** [comment]
ohh its basically 
- enrolling...and 
- if cannot enroll then use `--preview`, and 
- if that does not work then getting index file from friend who has enrolled already...

So yes, am able to get through it by enrolling those courses. Now guessing if someone maintained any repo of index pages for courses which are closed for enrollment and which does not allow preview.

**Comment by Mahesha999** [comment]
Just found this course: https://www.coursera.org/course/personalfinance
I guess it is not available for enrollment. So can we download it by any way?

**Comment by balta2ar** [comment]
Only if someone gave you the index page with the course Lectures. See this discussion for the details: https://github.com/coursera-dl/coursera-dl/issues/504#issuecomment-226945491

Maybe you could try this extension: https://chrome.google.com/webstore/detail/coursera-materials-downlo/ijkboagofaehocnjacacdhdcbbcpilih
But I think enrollment is still required for it to work.

**Comment by Mahesha999** [comment]
billion thanks for educating me!!!

**Comment by deepakjois** [comment]
I am seeing this on course `security` even after enrolling for it. This is `v2.ondemand` style course, so I was expecting it to work like all the other new self-paced courses. But the debug log says it is falling back to `download_old_style_class`. Any ideas?

```

08:38 $ ./coursera-dl  --debug -u <user> -p <pass> -- security
root[main] coursera_dl version 0.6.1
root[main] Downloading class: security
root[download_class] Downloading old style class security
root[get_cookies_from_cache] Trying to get cookies from the cache.
root[load_cookies_file] Loading cookie file /var/folders/y2/mys3_5bn16zdltnfx5d3dj3c0000gn/T/deepak_coursera_dl_cache/cookies/deepak.jois gmail.com.txt into memory.
root[get_cookies_from_cache] Loaded cookies from /var/folders/y2/mys3_5bn16zdltnfx5d3dj3c0000gn/T/deepak_coursera_dl_cache/cookies/deepak.jois@gmail.com.txt
requests.packages.urllib3.connectionpool[_new_conn] Starting new HTTPS connection (1): class.coursera.org
requests.packages.urllib3.connectionpool[_make_request] "HEAD /security/class HTTP/1.1" 302 0
root[validate_cookies] Stale session.
root[login] Initiating login.
root[login] There were no .coursera.org cookies to be cleared.
requests.packages.urllib3.connectionpool[_new_conn] Starting new HTTPS connection (1): class.coursera.org
requests.packages.urllib3.connectionpool[_make_request] "GET /security HTTP/1.1" 302 None
root[login] Forging cookie header: csrftoken=4OeCpgH3De6aZJxKsozt; csrf2_token_xN9j1cmP=7yD7MkbScU7UemL3EVg7z5vx.
requests.packages.urllib3.connectionpool[_new_conn] Starting new HTTPS connection (1): www.coursera.org
requests.packages.urllib3.connectionpool[_make_request] "POST /api/login/v3 HTTP/1.1" 200 None
root[login] Logged in on coursera.org.
requests.packages.urllib3.connectionpool[_make_request] "GET /security/auth/auth_redirector?type=login&subtype=normal HTTP/1.1" 302 None
requests.packages.urllib3.connectionpool[_make_request] "GET /course/security HTTP/1.1" 200 None
root[down_the_wabbit_hole] Following https://class.coursera.org/security/auth/auth_redirector?type=login&subtype=normal to authenticate on class.coursera.org.
root[down_the_wabbit_hole] Exiting "deep" authentication.
root[get_authentication_cookies] Found authentication cookies.
root[download_about] About file to be written to: security-about.json
root[download_about] About url: https://api.coursera.org/api/catalog.v1/courses?fields=largeIcon,photo,previewLink,shortDescription,smallIcon,smallIconHover,universityLogo,universityLogoSt,video,videoId,aboutTheCourse,targetAudience,faq,courseSyllabus,courseFormat,suggestedReadings,instructor,estimatedClassWorkload,aboutTheInstructor,recommendedBackground,subtitleLanguagesCsv&q=search&query=security
root[download_about] Downloading about page from: https://api.coursera.org/api/catalog.v1/courses?fields=largeIcon,photo,previewLink,shortDescription,smallIcon,smallIconHover,universityLogo,universityLogoSt,video,videoId,aboutTheCourse,targetAudience,faq,courseSyllabus,courseFormat,suggestedReadings,instructor,estimatedClassWorkload,aboutTheInstructor,recommendedBackground,subtitleLanguagesCsv&q=search&query=security
requests.packages.urllib3.connectionpool[_new_conn] Starting new HTTPS connection (1): api.coursera.org
requests.packages.urllib3.connectionpool[_make_request] "GET /api/catalog.v1/courses?fields=largeIcon,photo,previewLink,shortDescription,smallIcon,smallIconHover,universityLogo,universityLogoSt,video,videoId,aboutTheCourse,targetAudience,faq,courseSyllabus,courseFormat,suggestedReadings,instructor,estimatedClassWorkload,aboutTheInstructor,recommendedBackground,subtitleLanguagesCsv&q=search&query=security HTTP/1.1" 200 None
root[download_old_style_class] Subtitle unavailable in specified language
root[get_syllabus_url] Using index mode with page: https://class.coursera.org/security/lecture/index
requests.packages.urllib3.connectionpool[_make_request] "GET /security/lecture/index HTTP/1.1" 301 None
requests.packages.urllib3.connectionpool[_make_request] "GET /security/lecture HTTP/1.1" 302 None
requests.packages.urllib3.connectionpool[_make_request] "GET /course/security HTTP/1.1" 200 None
root[get_old_style_syllabus] Downloaded https://class.coursera.org/security/lecture/index (1162810 bytes)
root[parse_old_style_syllabus] Found 0 sections and 0 lectures on this page
root[parse_old_style_syllabus] The cookies file may be invalid, please re-run with the `--clear-cache` option.
```

**Comment by deepakjois** [comment]
Just an update on the `security` course specifically, I was able to force coursera-dl to download the course using `download_on_demand_class` method with the following patch: https://gist.github.com/deepakjois/caaeb10b28039d2a2c6888e81328a803

So there is at least one course which is not being detected properly as an on-demand course. Maybe the detection logic in `download_class` can be improved a little bit. Moreover, Coursera is moving over to the new platform anyway so we could default to downloading on-demand courses only by default.

**Comment by balta2ar** [comment]
Another course is `crypto`, see here: https://github.com/coursera-dl/coursera-dl/issues/508#issuecomment-226842459
After Coursera shuts down the old platform, we can safely remove support for the old-style courses, I think. As for now, old-style courses are preferred because people are rushing to save them.

**Comment by Mahesha999** [comment]
does this work for new courses too? In same way? Any tut? Especially what are class names which I can put at the end of command?

**Comment by fakeInsights** [comment]
Im also getting the same error when trying to download wharton-operations-analytics, a class I am enrolled.

I saw @deepakjois comment but I not sure how to make these changes using pip.

Thanks for your help.

coursera-dl -u <user> -p <password> -c cookie.txt --clear-cache --subtitle-language en wharton-operations-analytics
coursera_dl version 0.6.1
Downloading class: wharton-operations-analytics
Loaded cookies from uip.txt
Starting new HTTPS connection (1): class.coursera.org
Downloaded https://class.coursera.org/wharton-operations-analytics/lecture/index (1082 bytes)
Found 0 sections and 0 lectures on this page
The cookies file may be invalid, please re-run with the `--clear-cache` option.

**Comment by deepakjois** [comment]
> I saw @deepakjois comment but I not sure how to make these changes using pip.

@fakeInsights It is possible to make these changes if you are using a pip based installation as well. You need to locate where pip has installed coursera-dl source files on your system and then patch the `coursera_dl.py` file. However, it is not a very clean way to do it. You are messing with something that pip has installed.

Alternatively, you can install coursera-dl using the following method (from the README): https://github.com/coursera-dl/coursera-dl#alternative-installation-method-for-unix-systems

and then make the source code changes. These changes will be isolated in the virtualenv that you created, and anytime you need to go back to the standard pip version, you can just get out of the virtualenv.

**Comment by olemis** [comment]
I confirm that @deepakjois [comment](https://github.com/coursera-dl/coursera-dl/issues/512#issuecomment-227638981) works . Why isn't this implemented ?

**Comment by balta2ar** [comment]
This should be fixed in 0.7.0.
